src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorCSharpSemanticTokensInterceptor.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception;
+using Microsoft.VisualStudio.Utilities;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Export(typeof(MessageInterceptor))]
+    [InterceptMethod("workspace/semanticTokens/refresh")]
+    [ContentType(RazorLSPConstants.CSharpContentTypeName)]
+    internal class RazorCSharpSemanticTokensInterceptor : MessageInterceptor
+    {
+        private readonly LSPRequestInvoker _requestInvoker;
+
+        [ImportingConstructor]
+        public RazorCSharpSemanticTokensInterceptor(LSPRequestInvoker requestInvoker!!)
+        {
+            _requestInvoker = requestInvoker;
+        }
+
+        public override Task<InterceptionResult> ApplyChangesAsync(
+            JToken message, string containedLanguageName, CancellationToken cancellationToken)
+        {
+            var refreshParams = new SemanticTokensRefreshParams();
+            _ = _requestInvoker.ReinvokeRequestOnServerAsync<SemanticTokensRefreshParams, Unit>(
```

Comment:
I figured since it's a notification that we don't really care about awaiting the result. But the code is definitely cleaner if we `await` here instead, so maybe it's worth the tradeoff? I'll update it to your suggestion for now